### PR TITLE
Convert Redis operator to the new package structure

### PR DESCRIPTION
**Pull request by gkleiman**

### CODE REVIEW

File: repository/redis/0.1.0/framework.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: redis
+version: 0.1.0
+kudoVersion: 0.2.0
```
Comment (gerred) [inline repository/redis/0.1.0/framework.yaml]: make kudoVersion 0.3.0

**Review by gerred — CHANGES_REQUESTED** [review]

File: repository/redis/0.1.0/templates/init.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    spec:
+      restartPolicy: OnFailure
+      containers:
+      - name: {{ .PlanName }}
+        image: redis:5.0.1-alpine
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        - -c
+        - 'echo yes | redis-cli -x -h rediscluster1-redis-0.rediscluster1-svc.default.svc.cluster.local --cluster create --cluster-replicas 1 $(i=0; while [ $i -le 5 ]; do printf $(nslookup rediscluster1-redis-${i}.rediscluster1-svc.default.svc.cluster.local  | grep "Address" |  awk "{ print \$3 }")":6379 "; i=$((i+1)); done)'
```
Comment (fabianbaier) [inline repository/redis/0.1.0/templates/init.yaml]: For me this job fails when running:

```bash
$ go run cmd/kubectl-kudo/main.go install --instance=redis /Users/fabianbaier/go/src/github.com/kudobuilder/operators/repository/redis/0.1.0
```

output:
```bash
$ kubectl get pods -w
NAME                     READY   STATUS    RESTARTS   AGE
redis-deploy-job-rbksd   0/1     Pending   0          80m
redis-redis-0            1/1     Running   0          80m
redis-redis-1            1/1     Running   0          80m
redis-redis-2            1/1     Running   0          80m
redis-deploy-job-rbksd   0/1     ContainerCreating   0          80m
redis-deploy-job-rbksd   0/1     Error               0          80m
redis-deploy-job-rbksd   0/1     Error               1          80m
redis-deploy-job-rbksd   0/1     CrashLoopBackOff    1          80m
redis-deploy-job-rbksd   0/1     Error               2          80m
redis-deploy-job-rbksd   0/1     CrashLoopBackOff    2          81m
redis-deploy-job-rbksd   0/1     Error               3          81m
redis-deploy-job-rbksd   0/1     CrashLoopBackOff    3          81m
```

**Review by fabianbaier — CHANGES_REQUESTED** [review]
We might need to change the `init.yaml` along something like 

```
        - 'echo yes | redis-cli -x -h redis-redis-0 --cluster create --cluster-replicas 1 $(i=0; while [ $i -le 5 ]; do printf $(nslookup redis-redis-${i}  | grep "Address" |  awk "{ print \$3 }")":6379 "; i=$((i+1)); done)'
```

The output of the job could be a known issue with Alpine and nslookup https://github.com/gliderlabs/docker-alpine/issues/476#issuecomment-457124840 causing the job fail...

output:
```
$ kubectl logs redis-deploy-job-rbksd
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-0': Name does not resolve
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-1': Name does not resolve
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-2': Name does not resolve
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-3': Name does not resolve
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-4': Name does not resolve
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'redis-redis-5': Name does not resolve
Could not connect to Redis at :6379: Name does not resolve
```

File: repository/redis/0.1.0/operator.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: redis
+version: 0.1.0
+kudoVersion: 0.3.0
+maintainers:
```
Comment (fabianbaier) [inline repository/redis/0.1.0/operator.yaml]: @djannot was originally a maintainer/1stcommitter so would love to see him as well in this list :)

Comment (djannot) [reply repository/redis/0.1.0/operator.yaml]: Thanks for adding me, I'll take a look

**Comment by oliviabarrick** [comment]
Can you rebase to fix the tests?

**Review by oliviabarrick — APPROVED** [review]

**Review by fabianbaier — APPROVED** [review]

**Pull request merged**